packages/react-components/src/components/CaptionsSettingsModal.tsx

```diff
@@ -88,13 +92,13 @@ export const _CaptionsSettingsModal = (props: _CaptionsSettingsModalProps): JSX.
     if (isCaptionsFeatureActive) {
       onSetSpokenLanguage(language);
     } else {
-      onStartCaptions({ spokenLanguage: selectedItem.text });
+      onStartCaptions({ spokenLanguage: language });
```

Comment:
For all these language variable - we can call it languageCode in this component for better readability (cause we are mapping it to real language names)